includes/class-wc-stripe-intent-controller.php

```diff
@@ -1037,7 +1039,57 @@ public function create_and_confirm_setup_intent_ajax() {
 			wp_send_json_error(
 				[
 					'error' => [
-						'message' => $e->getLocalizedMessage(),
+						'message' => $e->getLocalizesdMessage(),
+					],
+				]
+			);
+		}
+	}
+
+	/**
+	 * Confirms the change payment method request for a subscription.
+	 *
+	 * This creates the payment method token from the setup intent.
+	 *
+	 * This function is used to confirm the change payment method request for a subscription after the user has been asked to authenticate their payment (eg 3D-Secure).
+	 * It is initiated from the subscription change payment method page.
+	 */
+	public function confirm_change_payment_from_setup_intent_ajax() {
+		try {
+			$is_nonce_valid = check_ajax_referer( 'wc_stripe_update_order_status_nonce', false, false );
+
+			if ( ! $is_nonce_valid ) {
+				throw new WC_Stripe_Exception( 'missing-nonce', __( 'CSRF verification failed.', 'woocommerce-gateway-stripe' ) );
+			}
+
+			if ( ! function_exists( 'wcs_is_subscription' ) ) {
+				throw new WC_Stripe_Exception( 'subscriptions_not_found', __( "We're not able to process this subscription change payment request payment. Please try again later.", 'woocommerce-gateway-stripe' ) );
+			}
+
+			$subscription_id = isset( $_POST['order_id'] ) ? absint( $_POST['order_id'] ) : false;
+			$subscription    = wcs_get_subscription( $subscription_id );
```

Comment:
`$subscription_id` can be `false` above. In this case, I think we should either:
1 - Make `$subscription` `false` as well (`$subscription = $subscription_id ? wcs_get_subscription( $subscription_id ) : false;`)
2 - Validate `$subscription_id` and throw an error as we do just below for `$subscription`
3 - Only load the subscription if `$subscription_id` is not `false`

Reply:
Thanks @wjrosa. I've gone with option 1 and only call `wcs_get_subscription()` if we have an ID. [4fa2fd1](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3313/commits/4fa2fd1a8dae406af5fc469d5eec04f2825ede4e)